tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -9,72 +9,89 @@
+        RetryTestOperation(testOp);
         Message("Test passed");
     }
 
-    operation RandomBit_Wrapper (throwawayMin : Int, throwawayMax : Int) : Int {
-        return RandomBit();
-    }
-
-
     // ------------------------------------------------------
     // Exercise 2.
     @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T2_RandomTwoBits () : Unit {
         Message("Testing two random bits generation...");
-        RetryCheckUniformDistribution(RandomTwoBits_Wrapper, 0, 3, 1000);
+        let userOp = RandomTwoBits;
+        let testUserOp = CheckUniformDistribution(_, (0,3), 1000); // testUserOp takes userOp of the form (Unit=>Int)
+        let testOp = Delay(testUserOp, userOp, _); // Delay() converts testUserOp to parameterless operation
+        RetryTestOperation(testOp);
         Message("Test passed");
     }
-
-    operation RandomTwoBits_Wrapper (throwawayMin : Int, throwawayMax : Int) : Int {
-        return RandomTwoBits();
-    }
     
 
     // ------------------------------------------------------
     // Exercise 3.
     @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T3_RandomNBits () : Unit {
         // Test random number generation for 1, 2, 3, 10 bits
-        for (min, max) in [(0, 1), (0, 3), (0, 7), (0, 1023)] {
-            let N = Ceiling( Lg(IntAsDouble(max + 1)) );
+        for N in [1, 2, 3, 10] {
             Message($"Testing N = {N}...");
-            RetryCheckUniformDistribution(RandomNBits_Wrapper, min, max, 1000);
+            let max = (1<<<N) - 1;
+            let userOp = Delay(RandomNBits, N, _); // Delay() converts RandomNBits to a parameterless operation
+            let testUserOp = CheckUniformDistribution(_, (0,max), 1000); // testUserOp takes userOp of the form (Unit=>Int)
+            let testOp = Delay(testUserOp, userOp, _); // Delay() converts testUserOp to parameterless operation
+            RetryTestOperation(testOp);
             Message($"Test passed for N = {N}");
 	    }
     }
 
-    operation RandomNBits_Wrapper (min : Int, max : Int) : Int {
-        // For N bit random number : min = 0, max = 2^N - 1
-        let N = Ceiling( Lg(IntAsDouble(max+1)) );
-        return RandomNBits(N);
-    }
-
 
     // ------------------------------------------------------
     /// # Summary
     /// Helper operation that checks that the given RNG operation generates a uniform distribution.
     /// # Input
-    /// ## f
+    /// ## op
     /// Random number generation operation to be tested.
-    /// ## min, max
+    /// ## range
     /// Minimal and maximal numbers in the range to be generated, inclusive.
     /// ## nRuns
     /// The number of random numbers to generate for test.
-    operation CheckUniformDistribution (f : ((Int, Int) => Int), min : Int, max : Int, nRuns : Int) : Bool {
+    operation CheckUniformDistribution (op : (Unit => Int), range : (Int,Int), nRuns : Int) : Bool {
```

Comment:
We should be able to keep min and max here rather than switching to range.

Reply:
Done